lib/modules/manager/pip-compile/artifacts.ts

```diff
@@ -1,152 +1,94 @@
-      }
-    }
+  const pipCompileArgs = extractHeaderCommand(content, outputFileName);
+  if (strict && pipCompileArgs.isCustomCommand) {
+    throw new Error(
+      'Detected custom command, header modified or set by CUSTOM_COMPILE_COMMAND',
+    );
   }
-  args.push(upath.parse(inputFileName).base);
-
-  return args.map((argument) => quote(argument)).join(' ');
+  return pipCompileArgs.argv.map(quote).join(' ');
 }
 
 export async function updateArtifacts({
   packageFileName: inputFileName,
   newPackageFileContent: newInputContent,
+  updatedDeps,
   config,
 }: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
-  const outputFileName = inputFileName.replace(regEx(/(\.in)?$/), '.txt');
-  logger.debug(
-    `pipCompile.updateArtifacts(${inputFileName}->${outputFileName})`,
-  );
-  const existingOutput = await readLocalFile(outputFileName, 'utf8');
-  if (!existingOutput) {
-    logger.debug('No pip-compile output file found');
+  if (!config.lockFiles) {
+    logger.warn(
+      { packageFileName: inputFileName },
+      'pip-compile: No lock files associated with a package file',
+    );
     return null;
   }
-  try {
-    await writeLocalFile(inputFileName, newInputContent);
-    if (config.isLockFileMaintenance) {
-      await deleteLocalFile(outputFileName);
-    }
-    const cmd = constructPipCompileCmd(
-      existingOutput,
-      inputFileName,
-      outputFileName,
-    );
-    const constraint = getPythonConstraint(config);
-    const pipToolsConstraint = getPipToolsConstraint(config);
-    const execOptions: ExecOptions = {
-      cwdFile: inputFileName,
-      docker: {},
-      toolConstraints: [
-        {
-          toolName: 'python',
-          constraint,
-        },
-        {
-          toolName: 'pip-tools',
-          constraint: pipToolsConstraint,
-        },
-      ],
-      extraEnv: {
-        PIP_CACHE_DIR: await ensureCacheDir('pip'),
-      },
-    };
-    logger.trace({ cmd }, 'pip-compile command');
-    await exec(cmd, execOptions);
-    const status = await getRepoStatus();
-    if (!status?.modified.includes(outputFileName)) {
+  logger.debug(
+    `pipCompile.updateArtifacts(${inputFileName}->${JSON.stringify(
+      config.lockFiles,
+    )})`,
+  );
+  const result: UpdateArtifactsResult[] = [];
+  for (const outputFileName of config.lockFiles) {
+    const existingOutput = await readLocalFile(outputFileName, 'utf8');
+    if (!existingOutput) {
+      logger.debug('pip-compile: No output file found');
       return null;
     }
-    logger.debug('Returning updated pip-compile result');
-    return [
-      {
+    try {
+      await writeLocalFile(inputFileName, newInputContent);
+      // TODO(not7cd): use --upgrade option instead deleting
+      if (config.isLockFileMaintenance) {
+        await deleteLocalFile(outputFileName);
+      }
+      const cmd = constructPipCompileCmd(existingOutput, outputFileName);
+      const execOptions: ExecOptions = await getExecOptions(
```

Comment:
```suggestion
      const execOptions = await getExecOptions(
```
should be inferred by compiler